### PR TITLE
add vsphere cluster creation

### PR DESCRIPTION
**Pull request by vxav**
### What does this PR do?

(Please set a descriptive PR title. Use this space for additional explanations.)

### What does it look like?

(If this is more than a textual change, a screenshot can help in some cases to speed up the review process.)

### Any background context you can provide?

(Please link public issues or summarize if not public.)

### Have you maintained the front matter?

(Please bump the last_review_date in case this qualifies as a review for an entire page. Provide user_questions which should be answered in the page. Provide a meaningful description.)

### CODE REVIEW

File: src/content/use-the-api/kubectl-gs/template-cluster.md [hunk 1]
```diff
@@ -178,10 +193,38 @@ It supports the following flags:
...
+- `--vsphere-worker-replicas` (optional) - Number of worker replicas (default: 3).
+- `--vsphere-resource-pool` (optional) - Name of the vSphere resource pool to deploy the VMs to (default: cluster)
+- `--vsphere-image-template` (optional) - Name of the vSphere template to deploy the VMs from (default: `ubuntu-2004-kube-${kubernetes-version}`).
+- `--vsphere-credentials-secret-name` (optional) - Name of the kubernetes secret that should be associated to the cluster app. It should exist in the organization's namespace and should contain the credentials for vsphere.
+- `--oidc-issuer-url` (optional) - This is the issuer URL for configuring OpenID connect in the cluster API.
+- `--oidc-ca-file` (optional) - This is the CA file path in case is not used a trusted Certificate Authority for OIDC endpoint.
+- `--oidc-client-id` (optional) - This is the client ID that is configured in the OIDC endpoint.
+- `--oidc-username-claim` (optional) - This is the claim used to map the username identity of the user.
+- `--oidc-groups-claim` (optional) - This is the claim used to map the group identity of the user.
```
Comment (pipo02mix) [inline src/content/use-the-api/kubectl-gs/template-cluster.md:222]: Are these vSphere specifics?

**Comment by pipo02mix** [comment]
can you update review date [here](https://github.com/giantswarm/docs/pull/1947/files#diff-a82ba408458f520c6135194014e4e4398544e3d4e8766a63064eeeed30bf99d4R20) too plz?

File: src/content/use-the-api/kubectl-gs/template-cluster.md (hunk 1 above)
Comment (vxav) [reply src/content/use-the-api/kubectl-gs/template-cluster.md:222]: No but they were in the openstack section and I don't see them in the other providers sections. Maybe they're just not up to date. Also I don't know if they apply to vintage.
Do you think we should move them to the general cli flags?

**Comment by vxav** [comment]
> can you update review date [here](https://github.com/giantswarm/docs/pull/1947/files#diff-a82ba408458f520c6135194014e4e4398544e3d4e8766a63064eeeed30bf99d4R20) too plz?

done in https://github.com/giantswarm/docs/pull/1947/commits/b04f9b17df6e645cede58a0896f38e3a14de8cef

File: src/content/use-the-api/kubectl-gs/template-cluster.md (hunk 1 above)
Comment (pipo02mix) [reply src/content/use-the-api/kubectl-gs/template-cluster.md:222]: I do think so, IMO oidc should be a general set of params for all CAPI clusters

Comment (vxav) [reply src/content/use-the-api/kubectl-gs/template-cluster.md]: WDYT https://github.com/giantswarm/docs/pull/1947/commits/4e3b37984f0dd0ed91ee44d2020a36e7cd14c7b7

**Review by pipo02mix — APPROVED** [review]
awesome

**Pull request merged**